src/main/host/descriptor/mod.rs

```diff
@@ -311,17 +331,19 @@ bitflags::bitflags! {
     }
 }
 
-#[derive(Clone)]
+#[derive(Clone, Debug)]
 pub struct Descriptor {
     file: Arc<AtomicRefCell<PosixFile>>,
     flags: DescriptorFlags,
+    count: Arc<()>,
```

Comment:
This seems unusual, and could use a comment.

I suppose we want to be able to get to the reference count given a reference to the `Descriptor` itself and not the `Arc` wrapping it?

Reply:
It's essentially a reference count for the descriptor object and not the file object. For example, if the plugin calls `fd1 = socket(AF_INET, ...); fd2 = dup(fd1);`, there will be two `Descriptor` objects pointing to the same file object. If the plugin then adds the socket `PosixFile` to an epoll instance, then there will still only be two `Descriptor` objects, but three references to the `PosixFile`. If the plugin then calls `close(fd1); close(fd2);`, both (all) `Descriptor` objects will be removed, but there will still be a reference to the file object in the epoll instance.

We want to run `close()` on the file object when all descriptors have been closed, even if there are still references to the `PosixFile` object.

If this make sense, I'll add a comment to explain what that `Arc` is being used for.

Reply:
Could you add a comment clarifying exactly what this count is? Sounds like it's the number of *open* Descriptors with a reference to the file?

Reply:
I've added some comments, and also renamed this to `open_count`.